### PR TITLE
[SHARE-388] [Feature] Update date parser

### PR DESCRIPTION
**Pull request by leb2dg**
* Improve date parser link
    * add upper and lower bounds
    * `None` causes `ValueError`
    * uses dateutil instead of Arrow to parse dates
    * breaks on bad dates
* Arrow --> Pendulum
* Add tests for date parser

### CODE REVIEW

File: share/normalize/links.py [hunk 1]
```diff
@@ -275,9 +276,18 @@ def execute(self, obj):
...
 
 class DateParserLink(AbstractLink):
+    LOWER_BOUND = pendulum.create(1200, 1, 1)
+    UPPER_BOUND = pendulum.now().add(years=100)
+    DEFAULT = pendulum.create(2016, 1, 1)
+
     def execute(self, obj):
         if obj:
-            return arrow.get(obj).to('UTC').isoformat()
+            date = dateutil.parser.parse(obj, default=self.DEFAULT)
```
Comment (chrisseto) [inline share/normalize/links.py]: Force the date into UTC here. If there isn't a timezone assume it is already UTC

File: tests/share/normalize/test_links.py [hunk 2]
```diff
@@ -1,8 +1,35 @@
...
 
-from share.normalize.links import DOILink
-from share.normalize.links import OrcidLink
+import calendar
+
+from share.normalize.links import DOILink, OrcidLink, DateParserLink
+
+
+@pytest.mark.parametrize('date, result', [
+    (None, None),
```
Comment (chrisseto) [inline tests/share/normalize/test_links.py]: `None` should raise an exception of some kind. We'll want to know if a normalizer is unknowingly passing along `None` values.

File: tests/share/normalize/test_links.py [hunk 3]
```diff
@@ -1,8 +1,35 @@
...
+    ('0059-11-01T00:00:00Z', ValueError('0059-11-01T00:00:00Z is before the lower bound 1200-01-01T00:00:00+00:00.')),
```
Comment (chrisseto) [inline tests/share/normalize/test_links.py]: Could you add a test case or two for coercing other time zones into UTC?
Maybe one for an invalid time zone as well?

File: share/normalize/links.py (hunk 1 above)
Comment (chrisseto) [inline share/normalize/links.py]: Tests are fine!

**Comment by leb2dg** [comment]
@chrisseto, comments (finally) addressed

**Pull request merged**